### PR TITLE
bpo-30557: faulthandler now correctly filters and displays exception codes on Windows

### PR DESCRIPTION
**Pull request by zooba**
bpo-30557: faulthandler now correctly filters and displays exception codes on Windows

### CODE REVIEW

File: Modules/faulthandler.c [hunk 1]
```diff
@@ -374,7 +374,7 @@ faulthandler_exc_handler(struct _EXCEPTION_POINTERS *exc_info)
     DWORD flags = exc_info->ExceptionRecord->ExceptionFlags;
 
     /* only log fatal exceptions */
```
Comment (vstinner) [inline Modules/faulthandler.c]: Please add bpo number to this comment.

**Review by vstinner — COMMENTED** [review]
Would it be possible to write an unit test for filtered exception? It's not a requirement, just a question.

**Comment by zooba** [comment]
@haypo Should be able to - the RaiseException function should let us raise with any code we like. Or perhaps I can figure out the call that brought this problem to light and make that one.

**Comment by vstinner** [comment]
Ah yes, using RaiseException() and check that faulthandler didn't log
anything is a good plan

**Comment by zooba** [comment]
Doesn't actually need a backport to 3.5, since the affected code doesn't appear to be there.

**Pull request merged**